### DISCUSSION TITLE
How to set multiple expression by oal

### DISCUSSION THREAD

**Discussion by 807801002** [apache/skywalking#7511] [Q&A]
Please answer these questions before submitting your issue.

- Why do you submit this issue?
- [x] Question or discussion
- [ ] Bug
- [ ] Requirement
- [ ] Feature or performance improvement

___
### Question
- What do you want to know?
https://github.com/apache/skywalking/blob/master/docs/en/concepts-and-designs/oal.md
The documents says that the filter expressions support linking by and, or and (...),I set the expression like: service_relation_redis_client_cpm = from(ServiceRelation.*).filter(detectPoint == DetectPoint.CLIENT and componentId == 7).cpm();  but it is failed.
![Snipaste_2021-08-20_16-53-55](https://user-images.githubusercontent.com/8719537/130208239-5038eb72-8ba2-4797-962d-0535da000c62.png)

___
### Bug
- Which version of SkyWalking, OS, and JRE?

- Which company or project?

- What happened?
If possible, provide a way to reproduce the error. e.g. demo application, component version.

___
### Requirement or improvement
- Please describe your requirements or improvement suggestions.

**Comment by wu-sheng** [comment]
`Filter` part should be write as a chain mode according to the grammar tree.

> service_relation_redis_client_cpm = from(ServiceRelation.*).filter(detectPoint == DetectPoint.CLIENT).filter(componentId == 7).cpm()

**Reply by 807801002** [reply]
Which field can be used at filter()？I thought all fields in org.apache.skywalking.oap.server.core.source.ServiceRelation can be used to the filter expression for "from(ServiceRelation.*)".But the "componentId == 7" is error. which is a field at ServiceRelation.
![Snipaste_2021-08-20_17-20-14](https://user-images.githubusercontent.com/8719537/130211520-ae6d3736-36f6-49a9-8f25-785f7d2f097d.png)

**Reply by love720720** [reply]
@wu-sheng 
Is there support for the OR expression syntax now? I expect to exclude the number of 404 HTTP requests in service_sla. However, I do not wish to exclude non-HTTP requests. How can this be achieved?

**Reply by wu-sheng** [reply]
There isn't or support. But usually we only tracing http, how do you trace others?

**Reply by love720720** [reply]
> There isn't or support. But usually we only tracing http, how do you trace others?

Sorry, I was mistaken in my understanding.
